index.d.ts

```diff
@@ -0,0 +1,373 @@
+// TypeScript Version: 4.2
+
+export type ImporterReturnType =
+  | {file: string}
+  | {contents: string}
+  | Error
+  | null;
+
+export type Importer = (
+  url: string,
+  prev: string,
+  done: (data: ImporterReturnType) => void
+) => ImporterReturnType | void;
+
+export interface LogData {
+  /**
+   * The message as-is in the file/data
+   */
+  message: string;
+
+  /**
+   * A set of information related to the location of the error
+   */
+  source: SourceSpan;
+
+  /**
+   * Is it a deprecation message
+   */
+  deprecation: boolean;
+}
+
+export interface Logger {
+  /**
+   * A function that is called each time `@warn` is encountered
+   * @param data All data relative to the `@warn` line
+   */
+  warn?(data: LogData): void;
+
+  /**
+   * A function that is called each time `@debug` is encountered
+   * @param data All data relative to the `@debug` line
+   */
+  debug?(data: LogData): void;
+}
+
+export interface Options {
```

Comment:
You'll probably need to split this into `Options` and `AsyncOptions` as well, which both inherit from `SharedOptions`.